dwst/scripts/test/particles.js

```diff
@@ -219,5 +233,77 @@ describe('particles module', () => {
         }).to.throw(InvalidParticles);
       });
     });
+    it('should throw InvalidParticles with correct errorparameters for broken expression start', () => {
+      expect(() => {
+        return parseParticles('foo$bar');
+      }).to.throw(InvalidParticles).that.does.include({
+        expression: 'foo$bar',
+        expected: '${',
+        expectedStringLength: '${'.length,
+        remainder: '$bar',
+      });
+    });
+    it('should throw InvalidParticles with correct errorparameters for escape at input end', () => {
```

Comment:
This case is missing from 'should throw an exception for invalid particles'. It should be added there unless that test is considered useless and removed entirely.

Reply:
It also seems that 'should throw an exception for invalid particles' contains some test cases that are missing from the new specific tests. Why is that? Are those test cases not relevant for testing the functionality?

Reply:
Fixed